common/djangoapps/course_modes/models.py

```diff
@@ -186,9 +186,13 @@ def modes_for_course(cls, course_id, only_selectable=True):
 
         """
         now = datetime.now(pytz.UTC)
-        found_course_modes = cls.objects.filter(
-            Q(course_id=course_id) & (Q(expiration_datetime__isnull=True) | Q(expiration_datetime__gte=now))
-        )
+        
+        found_course_modes = cls.objects.filter(course_id=course_id)
+        
+        if not include_expired:
```

Comment:
I think it would be helpful to add a comment that provides some context here.  It should explain what the old behavior was (excluding expired modes) and why we need expired mode information on the course details page.